### PR TITLE
Do not rewrite logs in rescue mode

### PR DESCRIPTION
**Pull request by jstodola**
When running in the rescue mode, anaconda should not rewrite installation logs on the system being rescued with the log files from the rescue environment.
Also do not change any SELinux contexts automatically on the system being rescued.

Resolves: RHEL-4794

### CODE REVIEW

**Comment by jstodola** [comment]
Just a note that I was not able to reproduce this problem on RHEL-10 or Rawhide, so porting is not necessary.

File: anaconda.py [hunk 1]
```diff
@@ -405,6 +405,11 @@ def _earlyExceptionHandler(ty, value, traceback):
     if ksdata.rescue.rescue:
         flags.rescue_mode = True
 
+    # Check if we are running in rescue mode to avoid running some post
+    # scripts later.
+    if flags.rescue_mode:
+        util.touch('/tmp/RESCUE_MODE')
```
Comment (jkonecny12) [inline anaconda.py:411]: Please move path to the file as variable to a `pyanaconda/core/constants.py`.

**Review by jkonecny12 — COMMENTED** [review]
This works nice. However, I wonder if passing environment variable wouldn't be more optimal solution than this. However, it would mean extending `execWithRedirect`:

https://github.com/rhinstaller/anaconda/blob/master/pyanaconda/core/util.py#L337

To be able to pass variable to `_run_program`. And then add the variable here:

https://github.com/rhinstaller/anaconda/blob/master/pyanaconda/kickstart.py#L115

Which might be unnecessary complicated, however, it's more polished solution then flag file.

@rvykydal @M4rtinK WDYT?

**Comment by jstodola** [comment]
An environment variable might be a nicer solution, but I got inspired by an already existing approach from https://github.com/rhinstaller/anaconda/commit/b7d699a547db725be17e01a7fa43d62725b9f8b5 and I didn't want to invent anything new. Also, this part of anaconda changed a lot by https://github.com/rhinstaller/anaconda/pull/3571, so this fix is really applicable just for RHEL-9.

File: anaconda.py (hunk 1 above)
Comment (jstodola) [reply anaconda.py]: Done

**Comment by jkonecny12** [comment]
Good point. If it is already used feel free to continue.

**Review by jkonecny12 — APPROVED** [review]

**Pull request merged**